default.nix

```diff
@@ -36,7 +36,7 @@ in
       tahoe-lafs = mach-nix.buildPythonPackage rec {
         inherit python providers;
         name = "tahoe-lafs";
-        version = "1.16.post999";
+        version = "1.17.0";
```

Comment:
I'm not sure I follow.  I guess the version declared here is the version that will be visible at runtime from the `allmydata` package because we're taking responsibility for making that part of things work.  I don't think anything automated will care what that version information says.  pip will care about the version in the packaging metadata.  Perhaps that comes from here as well?

From fiddling with the value and inspecting `mach-nix` and `pip` output it looks like maybe this number does also get used to populate that packaging metadata (presumably directly via the nix `version` attribute rather than by extracting it from `__init__.py` after we write it there).

Okay, so the packaging metadata might matter.  If we declare a requirement on, eg, `tahoe-lafs==1.17.3` then `1.17.0` isn't going to satisfy that so pip is going to break the build.  But `1.17.0.post999` also doesn't satisfy that so I'm not sure how that helps.

Right now we declare `>=1.17,<1.18` which is going to be happy with `1.17.0`, `1.17.0.post999`, or even `1.17.3`... So practically it seems like the packaging metadata version doesn't matter very much either (when it does, it seems like there's a good chance we'll be jumping to tahoe-lafs 1.18 and we'll have to bump a few different things).

What does seem like it would matter is a person reading a version number.  If we're shipping a revision of tahoe-lafs master from after 1.17.0 then it would be a bit misleading to declare we have exactly 1.17.0.  I can see 1.17.0.post999 being slightly better here because at least that signals it's "something after 1.17.0" (though it's probably not the implied 999 (or 1000?) somethings).

So it seems like it will "work" either way but "1.17.0.post999" might be less surprising to the human components.  Did I miss anything?

Reply:
> So it seems like it will "work" either way but "1.17.0.post999" might be less surprising to the human components.

Yeah, that is basically my reasoning. I picked `.post999` originally, because it seemed less likely to be confused with an actual tahoe-lafs version and more likely to prompt somebody to go "what where is that coming from" than assume they know what it means.